packages/services/src/competition.service.ts

```diff
@@ -1294,10 +1287,19 @@ export class CompetitionService {
     // Get leaderboard data for the competition to get scores and ranks
     const leaderboard = await this.getLeaderboard(competitionId);
     const leaderboardMap = new Map(
-      leaderboard.map((entry, index) => [
-        entry.agentId,
-        { score: entry.value, rank: index + 1 },
-      ]),
+      leaderboard.map((entry, index) => {
+        // If the agent's status in the competition is not active, score them as zero with a rank
+        // equal to the total number of agents (last place). This ensures rank-based sorting always
+        // works correctly.
+        if (
+          agents.find((agent) => agent.id === entry.agentId)
```

Comment:
Could use a  Set or Map for O(1) lookups here instead of the O(n × m) we have here